packages/affiliates/libs/uniswap/models.ts

```diff
@@ -0,0 +1,168 @@
+import assert from "assert";
+import { exists, getPositionKey } from "./utils";
+
+type Id = string;
+type MaybeId = {
+  id?: Id;
+};
+interface MakeId<T> {
+  (obj: T): Id;
+}
+
+// Generic table with common functions, store is a map now, but we could replace this with any kind of
+// data store with some minor modifications, as long as it has, set, get, delete, has and values
+export function Table<T extends MaybeId>(config: { makeId: MakeId<T>; type: string }, store: Map<Id, T>) {
+  const { makeId, type } = config;
+  async function create(data: T) {
+    const id = exists(data.id) ? data.id : makeId(data);
+    assert(!(await has(id)), `${type} exists`);
+    return set({ id, ...data });
+  }
+  async function set(data: T) {
+    assert(data.id, `${type} requires id`);
+    await store.set(data.id, { ...data });
+    return data;
+  }
+  async function get(id: string) {
```

Comment:
fixed both of these issues. The `set` function was fixed by changing `MaybeId` type to an interface rather than a type, then i could do `set(data: T & {id:Id})`.  Dont know why this works but it seems simplest.

also fixed a couple places where i was using string rather than Id